simd/unit_tests/TestSIMD.cpp

```diff
@@ -434,9 +459,28 @@ KOKKOS_INLINE_FUNCTION void device_check_condition() {
   checker.truth(all_of(a == decltype(a)(16)));
 }
 
+template <typename Abi>
+inline void host_check_math_ops_all_types(
+    Kokkos::Experimental::Impl::data_types<>) {}
```

Comment:
Prefer a fold expression to a recursive implementation.

Reply:
Changed in [6c197bc](https://github.com/kokkos/kokkos/pull/5913/commits/6c197bc0c8d74e590b116299bd74a8f0ef90e946).